(docs): Add interactive environments to Garden main repository (#4556)
garden-io/garden: README.md

![Garden dev deploy](https://raw.githubusercontent.com/ShankyJS/garden-quickstart-content/d8095ad1a8615edf49e721b8afcd901f3056e127/dev-mode.gif)

> Start using Garden right away by using any of the following two options:

Reviewer: We need to fix the sections a bit. 

With these changes it's

- Installation (H2)
- Demo (H2)
- Interactive environments (H3)
- Local Installation (H3)

So it's like interactive environments and local installation are nested under demo. Also "Usage" is missing. 

How about we instead do

-  Installation (H2)
- Demo (H2)
- Interactive Environments (H2)
- Usage (H2)

The goal with the usage section is first and foremost to show how it works at a glance, more so then be a follow up to the installation.

Author: I added this sections structure in latest commit, closing this comment for a new review.

- Make use of our brand new interactive environments that can help you deploy the project through interactive labs. ✅

> - You can also install the prerequisites in your computer and then follow the steps below. 💻

Reviewer: This kind of makes it sound like there's a choice between using interactive environments or installing Garden locally. 

However, interactive environments are for testing things out with an example project whereas you always need to install Garden locally to actually use it on your own projects. 

So let's make sure we don't conflate the two. I thing keeping the sections I suggested above more or less addresses that.

Author: Re-worked this.

### Interactive environments

Simply click any of the two buttons below ⬇️

<a href="https://go.garden.io/killercoda"><img src="https://raw.githubusercontent.com/garden-io/garden-interactive-environments/main/resources/img/killercoda-logo.png" alt="Killercoda logo in black and white." height="55px"/></a> [![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.svg)](https://go.garden.io/cloudshell)

> If you find any bugs 🐛 or have suggestions to improve our labs please don't hesitate in reaching out through an issue towards this [repository](https://github.com/garden-io/garden-interactive-environments/blob/main/resources/img/killercoda-logo.png) or ask through our [Discord Community](https://go.garden.io/discord)🌸

Reviewer: Change:

"hesitate in reaching out..."

to:

"hesitate to reach out by creating an [issue here](https://github.com/garden-io/garden-interactive-environments/issues) or by asking in [our Discord Community](https://go.garden.io/discord) 🌸

![Garden dev deploy](https://raw.githubusercontent.com/ShankyJS/garden-quickstart-content/d8095ad1a8615edf49e721b8afcd901f3056e127/dev-mode.gif)

## Interactive environments

> If you want to try Garden in an interactive cloud-based environment with a playground-like experience, use our new Killercoda and CloudShell environments ✨.

Author: Brand new text optimized with Grammarly and ChatGPT for potential grammar issues; I hope it's ok and less wordy than the first try. 

cc @worldofgeese / @eysi09

> If you want to try Garden in an interactive cloud-based environment with a playground-like experience, use our new Killercoda and CloudShell environments ✨.

Reviewer 2: ```suggestion
Preview Garden with our new interactive and install-free cloud-based playgrounds ✨.
```

> Click on either of the two buttons below ⬇️.

Reviewer 2: ```suggestion
Click a button to start your Killercoda or Google Cloud Shell environment 👇🏼.
```

<a href="https://go.garden.io/killercoda"><img src="https://raw.githubusercontent.com/garden-io/garden-interactive-environments/main/resources/img/killercoda-logo.png" alt="Killercoda logo in black and white." height="55px"/></a> [![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.svg)](https://go.garden.io/cloudshell)

> If you find any bugs 🐛 or have suggestions to improve our labs please don't hesitate to reach out by creating an [issue here](https://github.com/garden-io/garden-interactive-environments) or by asking in our [Discord Community](https://go.garden.io/discord)🌸

Author: Apply change requested by @eysi09!

## Usage

> > Make sure you have Garden installed and Kubernetes running locally (e.g. with Minikube or Docker for Desktop) before deploying the project.

Author: I brought this line here as we don't specify this in the document, which adds good value to it IMO.

cc @eysi09
